### PR TITLE
build(buildscript):: add proxy example for windows

### PR DESCRIPTION
**Pull request by Imitater967**
I believe this will help people setup proxy easier

### CODE REVIEW

File: gradlew.bat [hunk 1]
```diff
@@ -35,6 +35,8 @@ for %%i in ("%APP_HOME%") do set APP_HOME=%%~fi
 
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
 set DEFAULT_JVM_OPTS="-Xmx64m" "-Xms64m"
+% set GRADLE_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890 -Dhttps.proxyHost=127.0.0.1 -Dhttps.proxyPort=7890 %
+% #set JAVA_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890 %
```
Comment (BenjaminAmos) [inline gradlew.bat]:
```suggestion
@rem set GRADLE_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890 -Dhttps.proxyHost=127.0.0.1 -Dhttps.proxyPort=7890
@rem set JAVA_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890
```

File: groovyw.bat [hunk 2]
```diff
@@ -34,6 +34,8 @@ for %%i in ("%APP_HOME%") do set APP_HOME=%%~fi
 
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
 set DEFAULT_JVM_OPTS=
+% #set GRADLE_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890 -Dhttps.proxyHost=127.0.0.1 -Dhttps.proxyPort=7890 %
+% #set JAVA_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890 %
```
Comment (BenjaminAmos) [inline groovyw.bat]:
```suggestion
@rem set GRADLE_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890 -Dhttps.proxyHost=127.0.0.1 -Dhttps.proxyPort=7890
@rem set JAVA_OPTS=-Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=7890
```

**Review by BenjaminAmos — COMMENTED** [review]
These are changes for a very specific scenario which I am not aware of any other contributors having to use. Intercepting proxies can be harmful when misused and whlist they have some very legitimate purposes, I would not promote their use casually. If a user requires the use of a proxy, they should either have this configured system-wide or should be knowledgeable enough to configure the programs they use to accept it. As such, I would not be comfortable accepting this change personally. I will leave this open though as others may hold different opinions.

In addition, as far as I know the correct syntax for comments in batch files is to use the `REM` command (it may stand for "remark").

**Comment by Imitater967** [comment]
as i know, the github in my country have a ultra slow downloading speed. that's why we need a proxy.
the java/gradle don't accept the global proxy, until you adding -Dhttp.proxy.

in my country, we need proxy even for Android Studio, Spring boot, and any gradle project.

Trust me , proxy is very useful for some continent

**Comment by jdrueckert** [comment]
> Trust me , proxy is very useful for some continent

In that case, this should be a country-specific note in the troubleshooting section of our docs, but not integrated in the scripts by default.

**Comment by soloturn** [comment]
a proxy is a standard configuration and should not go into gradlew, this is updated by gradlew update and wiped out. proxy can and should be set into the users or system environment, as HTTP_PROXY and HTTPS_PROXY. if this does not work, set GRADLE_OPTS and JAVA_OPTS into the users environment. if this is still not enough, please use something like direnv: https://direnv.net/ which exists for windows as well. if you feel this should go somewhere into the documentation, please change one of the doc files and reopen.

**Pull request closed without merging**